docs/docs/module_guides/indexing/lpg_index_guide.md

```diff
@@ -0,0 +1,573 @@
+- `graph_store.delete(properties={"key": "val"})` - delete based on properties
+- `graph_store.structured_query("<cypher query>")` - runs a cypher query (assuming the graph store supports it)
+
+In addition `a` versions exist for all of these for async support (i.e. `aget`, `adelete`, etc.).
+
+## Advanced Customization
+
+As with all components in LlamaIndex, you can sub-class modules and customize things to work exactly as you need, or try out new ideas and research new modules!
+
+### Sub-Classing Extractors
+
+Graph extractors in LlamaIndex subclass the `TransformComponent` class. If you've worked with the ingestion pipeline before, this will be familiar since it is the same class.
+
+The requirement for extractors is that the insert graph data into the metadata of the node, which will then be processed later on by the index.
+
+Here is a small example of sub-classing to create a custom extractor:
+
+```python
+from llama_index.core.graph_store.types import (
+    EntityNode,
+    Relation,
+    KG_NODES_KEY,
+    KG_RELATIONS_KEY,
+)
+from llama_index.core.schema import BaseNode, TransformComponent
+
+
+class MyGraphExtractor(TransformComponent):
+    # the init is optional
+    # def __init__(self, ...):
+    #     ...
+
+    def __call__(
+        self, llama_nodes: list[BaseNode], **kwargs
+    ) -> list[BaseNode]:
+        for llama_node in llama_nodes:
+            # be sure to not overwrite existing entities/relations
+
+            existing_nodes = llama_node.metadata.pop(KG_NODES_KEY, [])
+            existing_relations = llama_node.metadata.pop(KG_RELATIONS_KEY, [])
+
+            existing_nodes.append(
+                EntityNode(
+                    name="llama", label="ANIMAL", properties={"key": "val"}
+                )
+            )
+            existing_nodes.append(
+                EntityNode(
+                    name="index", label="THING", properties={"key": "val"}
+                )
+            )
+
+            existing_relations.append(
+                Relation(
+                    label="HAS",
+                    source_id="llama",
+                    target_id="index",
+                    properties={},
+                )
+            )
+
+            # add back to the metadata
+
+            llama_node.metadata["nodes"] = existing_nodes
+            llama_node.metadata["relations"] = existing_relations
+
+        return llama_nodes
+
+    # optional async method
+    # async def acall(self, llama_nodes: list[BaseNode], **kwargs) -> list[BaseNode]:
+    #    ...
+```
+
+### Sub-Classing Retrievers
+
+The retriever is a bit more complicated than the extractors, and has it's own special class to help make sub-classing easier.
+
+Here is a small example of sub-classing to create a custom retriever:
+
+```python
+from llama_index.core.indices.property_graph import CustomPGRetriever
+
+
+class MyCustomRetriever(CustomPGRetriever):
+    def init(my_option_1: bool = False, **kwargs) -> None:
```

Comment:
- nit: missing "self" 
- broader note: I think we might want to consider explicitly passing in graph_store as a param (so the user always needs to specify graph_store as the first param, and include_text as the second param), so the user can do stuff with the graph store if they want